hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ChecksumFileSystem.java

```diff
@@ -286,8 +295,123 @@ protected int readChunk(long pos, byte[] buf, int offset, int len,
     public IOStatistics getIOStatistics() {
       return IOStatisticsSupport.retrieveIOStatistics(datas);
     }
+
+    public static long findChecksumOffset(long dataOffset,
```

Comment:
does bytesPerSum ever come in as 0?

Reply:
It is written in the cheksum file as the first int using `sums.writeInt(bytesPerSum);`
Default value is 512. 
If reading this value  causes some exception it defaults to 1. 
So I don't think it can ever come as 0.